File: src/073. Counting fractions in a range/073.rb
require 'set'

fractions = Set.new
1.upto(12_000) do |d|
  (d / 3 + 1).upto(d / 2) do |n|
    fractions.add(n.quo(d))
  end
end

puts fractions.size - 1 # remove the number 1/2

File: src/073. Counting fractions in a range/073.py
from fractions import Fraction

fractions = set()
for d in range(1, 12001):
    for n in range(d // 3 + 1, d / 2):
        fractions.add(Fraction(n, d))

print(len(fractions) - 1) # remove the number 1/2
